tests/components/yamaha/test_media_player.py

```diff
@@ -23,54 +26,142 @@ def __init__(self, ctrl_url, name, zones=None):
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is None
+
+
+async def test_enable_output(hass, device, main_zone):
+    """Test enable output service."""
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+
+    port = "hdmi1"
+    enabled = True
+    data = {
+        "entity_id": "media_player.yamaha_receiver_main_zone",
+        "port": port,
+        "enabled": enabled,
+    }
+
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_ENABLE_OUTPUT, data, True)
+
+    assert main_zone.enable_output.call_count == 1
+    assert main_zone.enable_output.call_args == call(port, enabled)
+
+
+async def test_select_scene(hass, device, main_zone, caplog):
+    """Test select scene service."""
+    scene_prop = PropertyMock(return_value=None)
+    type(main_zone).scene = scene_prop
+
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+
+    scene = "TV Viewing"
+    data = {
+        "entity_id": "media_player.yamaha_receiver_main_zone",
+        "scene": scene,
+    }
 
-    def tearDown(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_SELECT_SCENE, data, True)
 
-    def enable_output(self, port, enabled):
-        """Enable output on a specific port."""
-        data = {
-            "entity_id": "media_player.yamaha_receiver_main_zone",
-            "port": port,
-            "enabled": enabled,
-        }
+    assert scene_prop.call_count == 1
+    assert scene_prop.call_args == call(scene)
 
-        self.hass.services.call(yamaha.DOMAIN, yamaha.SERVICE_ENABLE_OUTPUT, data, True)
+    scene = "BD/DVD Movie Viewing"
+    data["scene"] = scene
 
-    def create_receiver(self, mock_rxv):
-        """Create a mocked receiver."""
-        mock_rxv.return_value = self.device
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_SELECT_SCENE, data, True)
 
-        config = {"media_player": {"platform": "yamaha", "host": "127.0.0.1"}}
+    assert scene_prop.call_count == 2
+    assert scene_prop.call_args == call(scene)
 
-        assert setup_component(self.hass, mp.DOMAIN, config)
-        self.hass.block_till_done()
+    scene_prop.side_effect = AssertionError()
 
-    @patch("rxv.RXV")
-    def test_enable_output(self, mock_rxv):
-        """Test enabling and disabling outputs."""
-        self.create_receiver(mock_rxv)
+    missing_scene = "Missing scene"
+    data["scene"] = missing_scene
 
-        self.enable_output("hdmi1", True)
-        self.main_zone.enable_output.assert_called_with("hdmi1", True)
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_SELECT_SCENE, data, True)
 
-        self.enable_output("hdmi2", False)
-        self.main_zone.enable_output.assert_called_with("hdmi2", False)
+    assert f"Scene '{missing_scene}' does not exist!" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5662368)